### PR TITLE
Better implementation for supporting Play database evolutions. Fix #245

### LINKED ISSUE #245: Don't use play.modules.disabled
`play.modules.disabled` is intended for end users to be able to override what modules are enabled by default. If libraries provide config for it, this undermines what the user can do - for example, if the user for whatever reason has play-slick on the classpath, but they want it disabled and Play's built in DBModule enabled, then they can't do that, because slick has already provided default configuration that disables Play's built in DBModule.

Additionally, since Play by default provides an empty list for the disabled modules, this means the current configuration is susceptible to classpath ordering issues, if Play's jar is loaded after play-slick, then play-slicks disabled modules config will be overridden by the empty list that Play provides.

### PR DESCRIPTION
**Pull request by dotta**
After having split play-jdbc in different modules, the Play Slick module can depend
on play-jdbc-api alone, allowing it to bind `DBApi` to a different implementation than
the one provided by `play-jdbc`. The clear benefit of this change is that we no longer
need to disable modules in the reference.conf, because both the `DBModule` and `HikariCPModule` are no longer in the classpath.

Furthermore, evolutions support was moved out of the Play Slick core, so that it
is now an opt-in feature.

This commit depends on playframework/playframework#4380

### CODE REVIEW

**Comment by dotta** [comment]
This PR won't compile until playframework/playframework#4380 is merged, and a nightly build based on it is made available.

**Comment by dotta** [comment]
This is ready for review.

@jroper Could you have a look at the build, making sure that I haven't broke anything (in particular, I am thinking about OmniDoc).

File: docs/manual/working/scalaGuide/main/sql/slick/PlaySlick.md [hunk 1]
```diff
@@ -11,6 +11,14 @@ Play Slick currently supports Slick 3.0 with Play 2.4, for both Scala 2.10 and 2
 
 > Note: This guide assumes you already know both Play 2.4 and Slick 3.0. Also, it assumes you will be using the new Slick Database I/O Actions API. In fact, using the deprecated Invoker/Execution API of Slick should be possible, but it might not be as convenient. Also, be aware that the Invoker/Execution API is planned to be removed in the next major release of Slick.
 
+### Getting Help
+
+If you are having trouble using Play Slick, check if the [[FAQ|PlaySlickFAQ]] contains the answer. Otherwise, feel free to reach out to [play-framework user group] for any question.
+Also, note that if you are seeking help on Slick, the [slick user group] may be a better place to get support.
+
+[play-framework user group]: https://groups.google.com/forum/#!forum/play-framework
+[slick user group]: https://groups.google.com/forum/#!forum/scalaquery
```
Comment (richdougherty) [inline docs/manual/working/scalaGuide/main/sql/slick/PlaySlick.md]: Isn't there a company that provides commercial support? ;)

File: docs/manual/working/scalaGuide/main/sql/slick/PlaySlickFAQ.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Play Slick FAQ
```
Comment (richdougherty) [inline docs/manual/working/scalaGuide/main/sql/slick/PlaySlickFAQ.md]: Good idea to add a FAQ.

**Comment by richdougherty** [comment]
LGTM. :+1: Did you want to squash the commits or is this ready for merge as it is?

**Comment by dotta** [comment]
@richdougherty I've just squashed one commit, and this PR is now ready to be merged. I'd like to keep the documentation commit separated from the one refactoring the modules, if you don't mind.

**Comment by richdougherty** [comment]
@dotta, it's fine to have more than one commit so long as there's a reason for it, like in this case.

**Comment by dotta** [comment]
:fireworks: Thanks @richdougherty for merging!

File: docs/manual/working/scalaGuide/main/sql/slick/PlaySlick.md (hunk 1 above)
Comment (dotta) [inline docs/manual/working/scalaGuide/main/sql/slick/PlaySlick.md]: Good point. Here you go https://github.com/playframework/play-slick/pull/252

**Pull request merged**